contracts/utils/SAVAXDepositHelper.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault} from "../interfaces/IRibbonVault.sol";
+
+contract SAVAXDepositHelper {
+    ISAVAX public sAVAX;
+    IRibbonVault public sAVAXVault;
+
+    constructor(address _sAVAX, address _sAVAXVault) {
+        require(_sAVAX != address(0), "!sAVAX");
+        require(_sAVAXVault != address(0), "!sAVAXVault");
+
+        sAVAX = ISAVAX(_sAVAX);
+        sAVAXVault = IRibbonVault(_sAVAXVault);
+    }
+
+    function deposit() external payable {
+        uint256 sAVAXAmount = sAVAX.submit{value: msg.value}();
```

Comment:
Do we have to cover the edge case where `sAVAXAmount` is not equal to `msg.value` here? Since the user sends in the `msg.value` there may be cases where it is larger than `sAVAXAmount` and so some dust gets stuck in the contract.

We'll have to do a refund in that scenario:

```sol
uint256 sAVAXAmount = sAVAX.submit{value: msg.value}();

if (msg.value > sAVAXAmount) {
  (bool success, ) = msg.sender.call{value:msg.value-sAVAXAmount}("");
  require(success, "Transfer failed.");
}
```

Reply:
`sAVAXAmount` is less than `msg.value`, never equal or greater.
I suppose we could add a check? But I dunno if this is required

Reply:
Maybe if their validator gets slashed too many times? `sAVAXAmount` could be greater than `msg.value`?  I don't really think these scenarios will happen.

Reply:
sAVAX is not 1:1 with AVAX, its a share-based receipt so sAVAX will always be < msg.value (AVAX amount deposited).